ocs_ci/ocs/bucket_utils.py

```diff
@@ -139,6 +140,23 @@ def retrieve_test_objects_to_pod(podobj, target_dir):
     return downloaded_objects
 
 
+@retry(UnexpectedBehaviour, tries=5, delay=10, backoff=1)
+def check_objects_in_bucket(bucket_name, objects_list, mcg_obj, s3pod):
```

Comment:
The check in `test_namespace_crd.py::test_respin_mcg_pod_and_check_data_integrity_crd` line 997 expects a potential false value that this won't return even if the bucket doesn't contain the objects - you need to return False instead of raising an exception here.  

However, the way you're currently using the retry function / decorator doesn't allow it. Also, because the timeout technicals are set in the decorator, this doesn't allow future users of this function to decide whether they want to wait 10 minutes instead of the ~1 minute that's currently set.

Please use retry (as a function and not a decorator) or the TimeoutSampler class on an anonymous function and return a boolean depending on the result, and add the necessary arguments to allow configuring the time.

Reply:
For example:

```
def is_remote_number_even(url, timeout=60):
    """
    Check if the number retrieved over HTTP is even.
     
    Args:
        timeout (int): Timeout in seconds
  
    Returns:
         bool: True if number is even, False otherwise
        
    """"
    def _is_remote_number_even(url_arg):
         return http_call_result(url_arg) % 2 == 0
     
    return any(result for result in TimeoutSampler(timetout, 10, _is_remote_number_even, url))          
```

Reply:
@sagihirshfeld Even if we try to use TimeoutSampler class, it will raise exception and it will not return false value once timeout gets completed